types/p-map/index.d.ts

```diff
@@ -0,0 +1,44 @@
+                                                                  options?: pMap.Options): Promise<[M, M, M, M, M, M, M, M, M, M]>;
+declare function pMap<T1, T2, T3, T4, T5, T6, T7, T8, T9, M>(input: [Input<T1>, Input<T2>, Input<T3>, Input<T4>, Input<T5>, Input<T6>, Input<T7>, Input<T8>, Input<T9>],
+                                                             mapper: Mapper<T1 | T2 | T3 | T4 | T5 | T6 | T7 | T8 | T9, M>,
+                                                             options?: pMap.Options): Promise<[M, M, M, M, M, M, M, M, M]>;
+declare function pMap<T1, T2, T3, T4, T5, T6, T7, T8, M>(input: [Input<T1>, Input<T2>, Input<T3>, Input<T4>, Input<T5>, Input<T6>, Input<T7>, Input<T8>],
+                                                         mapper: Mapper<T1 | T2 | T3 | T4 | T5 | T6 | T7 | T8, M>,
+                                                         options?: pMap.Options): Promise<[M, M, M, M, M, M, M, M]>;
+declare function pMap<T1, T2, T3, T4, T5, T6, T7, M>(input: [Input<T1>, Input<T2>, Input<T3>, Input<T4>, Input<T5>, Input<T6>, Input<T7>],
+                                                     mapper: Mapper<T1 | T2 | T3 | T4 | T5 | T6 | T7, M>,
+                                                     options?: pMap.Options): Promise<[M, M, M, M, M, M, M]>;
+declare function pMap<T1, T2, T3, T4, T5, T6, M>(input: [Input<T1>, Input<T2>, Input<T3>, Input<T4>, Input<T5>, Input<T6>],
+                                                 mapper: Mapper<T1 | T2 | T3 | T4 | T5 | T6, M>,
+                                                 options?: pMap.Options): Promise<[M, M, M, M, M, M]>;
+declare function pMap<T1, T2, T3, T4, T5, M>(input: [Input<T1>, Input<T2>, Input<T3>, Input<T4>, Input<T5>],
+                                             mapper: Mapper<T1 | T2 | T3 | T4 | T5, M>,
+                                             options?: pMap.Options): Promise<[M, M, M, M, M]>;
+declare function pMap<T1, T2, T3, T4, M>(input: [Input<T1>, Input<T2>, Input<T3>, Input<T4>],
+                                         mapper: Mapper<T1 | T2 | T3 | T4, M>,
+                                         options?: pMap.Options): Promise<[M, M, M, M]>;
+declare function pMap<T1, T2, T3, M>(input: [Input<T1>, Input<T2>, Input<T3>],
+                                     mapper: Mapper<T1 | T2 | T3, M>,
+                                     options?: pMap.Options): Promise<[M, M, M]>;
+declare function pMap<T1, T2, M>(input: [Input<T1>, Input<T2>], mapper: Mapper<T1 | T2, M>, options?: pMap.Options): Promise<[M, M]>;
+declare function pMap<T1, M1>(input: [Input<T1>], mapper: Mapper<T1, M1>, options?: pMap.Options): Promise<[M1]>;
+declare function pMap(input: Iterable<Input<any>>, mapper: Mapper<any, any>, options?: pMap.Options): Promise<any[]>;
+
+type Input<T> = Promise<T> | PromiseLike<T> | T;
+type Mapper<T, R> = (el: T, index: number) => Promise<R> | R;
```

Comment:
But with the above definition if the array if of different types, T inferred/can be specified will be union of those 5 types